lib/finch/http1/conn.ex

```diff
@@ -108,26 +108,80 @@ defmodule Finch.Conn do
 
     start_time = Telemetry.start(:request, metadata, extra_measurements)
 
-    case HTTP1.request(conn.mint, req.method, full_path, req.headers, req.body) do
+    case HTTP1.request(conn.mint, req.method, full_path, req.headers, stream_or_body(req.body)) do
       {:ok, mint, ref} ->
-        Telemetry.stop(:request, start_time, metadata, extra_measurements)
-        start_time = Telemetry.start(:response, metadata, extra_measurements)
-
-        case receive_response([], acc, fun, mint, ref, receive_timeout) do
-          {:ok, mint, acc} ->
-            Telemetry.stop(:response, start_time, metadata, extra_measurements)
-            {:ok, %{conn | mint: mint}, acc}
+        case maybe_stream_request_body(mint, ref, req.body, receive_timeout) do
+          {:ok, mint} ->
+            Telemetry.stop(:request, start_time, metadata, extra_measurements)
+            start_time = Telemetry.start(:response, metadata, extra_measurements)
+
+            case receive_response([], acc, fun, mint, ref, receive_timeout) do
```

Comment:
I would like to try to flatten this out and avoid 3 levels of nested cases. Maybe that could be as simple as adding `receive_response/5`, leaving out the first argument?

Reply:
good idea

Reply:
@sneako actually I added track_request_telemetry funuction instead, which wraps the response handling and is responsible for what the name suggests instead. Otherwise we'd have to pass around even more parameters as extra measurements are needed there too. And this way we have separation of concerns in place here.

There's a closure / anonymous function being passed as a last argument, however. I don't think it's a problem but let me know if it is.